debugtools/DDR_VM/src/com/ibm/j9ddr/view/dtfj/image/J9DDRImageProcess.java

```diff
@@ -578,24 +566,59 @@ public J9DDRBaseImageThread getThread(long id) {
 		Map<Long, Object> dtfjThreadMap = getThreadMap();
 		return (J9DDRBaseImageThread)dtfjThreadMap.get(id);
 	}
-	
+
 	private int resolveGenericSignal(int num) {
-		if ((num & J9PORT_SIG_FLAG_SIGQUIT) != 0) 	return 3;
-		if ((num & J9PORT_SIG_FLAG_SIGILL) != 0) 	return 4;
-		if ((num & J9PORT_SIG_FLAG_SIGTRAP) != 0) 	return 5;
-		if ((num & J9PORT_SIG_FLAG_SIGABRT) != 0) 	return 6;
-		if ((num & J9PORT_SIG_FLAG_SIGFPE) != 0) {
-			if (num == J9PORT_SIG_FLAG_SIGFPE_DIV_BY_ZERO) 		return 35;
-			if (num == J9PORT_SIG_FLAG_SIGFPE_INT_DIV_BY_ZERO) 	return 36;
-			if (num == J9PORT_SIG_FLAG_SIGFPE_INT_OVERFLOW) 	return 37;
+		Platform os = process.getPlatform();
+
+		if ((num & OMRPORT_SIG_FLAG_SIGQUIT) != 0) {
+			if (os == Platform.ZOS) return 24;
+			if (os == Platform.WINDOWS) return 21;
+			return 3;
+		}
```

Comment:
> I don't think we need the OS tests (notice "Generic" in the method name)

I thought the purpose of this method was to return the signal that caused a crash given the generic signal number reported by the crash:

```java
    /**
     * Get the OS signal number in this process which triggered the creation 
     * of this image.
     * @return the OS signal number in this process which triggered the creation 
     * of this image, or 0 if the image was not created because of a signal in this
     * process
     * 
     * @exception DataUnavailable if the information cannot be provided
     * @throws CorruptDataException 
     */
    public int getSignalNumber() throws DataUnavailable, CorruptDataException;
```

I think this function has been incorrect for some time and the OS-specific logic is needed to convert the generic signal number into the real signal number that drove the original crash.

Reply:
No, as I read the code, it is intended to map a 'Generic' signal (as it appears in the javacore file) to an index into the `SIGNAL_NAMES` array: the resulting value is not OS-specific. Yes, it has been broken for a while (since eclipse/omr#3921), but we're fixing that now.